pkg/validation/application_definition.go

```diff
@@ -94,6 +97,72 @@ func validateSource(source appskubermaticv1.ApplicationSource, f *field.Path) []
 	return allErrs
 }
 
+func validateHelmSource(helmSource *appskubermaticv1.HelmSource, f *field.Path) field.ErrorList {
+	allErrs := validateHelmSourceURL(helmSource, f)
+
+	if e := validateHelmCredentials(helmSource.Credentials, f.Child("credentials")); e != nil {
+		allErrs = append(allErrs, e)
+	}
+
+	return allErrs
+}
+
+func validateHelmSourceURL(helmSource *appskubermaticv1.HelmSource, f *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// url.Parse is _extremely_ forgiving and happily accepts nonsense like "[" or "123" or even "'"
+	// as valid URLs.
+	parsed, err := url.Parse(helmSource.URL)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(f.Child("url"), helmSource.URL, err.Error()))
+		return allErrs
+	}
+
+	if parsed.Host == "" || parsed.Scheme == "" {
+		allErrs = append(allErrs, field.Invalid(f.Child("url"), helmSource.URL, "value must be a valid URL"))
+		return allErrs
+	}
+
+	isLocalhost, err := docker.MatchLocalhost(parsed.Host)
```

Comment:
nit: I think it makes sense here to have a short comment that we are doing this check because containerd will automatically configure any request with localhost to use http